### PR TITLE
fix(AlertReportModal): Text Area Change

### PR DESCRIPTION
**Pull request by AAfghahi**
### SUMMARY

There was an error report where users were reporting that sql values that they were inputting into the TextAreaControl section would linger even when they closed out and went to a new report, or even created a new report. This was because we were storing the sql in state, but not clearing state when a modal was closed. 

This state was created to stop cursor errors here: https://github.com/apache/superset/pull/15820. I believe that @geido made this change. 

The cursor errors were occurring because we added a debouncer to the ace-editor here: https://github.com/apache/superset/pull/10837

In investigating this, I was wondering why we had a debouncer for the ace editor to begin with? The input is not using an API, and looking through the different places where TextAreaControl is used, none of them make an API call. So this PR deletes the debouncer, which fixes both the cursor back issue and also gets rid of the state.value //prop.value difference. 

However, I wanted to make sure that getting rid of this debouncer is fine @ktmud and @geido ? Where are the places that we should check to make sure?

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

Before:

https://user-images.githubusercontent.com/48933336/138182082-471e1948-0591-4f86-a4b8-21c264a41ff3.mp4

After:
https://user-images.githubusercontent.com/48933336/138182251-f5d5dfc0-c7dc-4066-a39a-b1b2b1b0cffd.mov

### TESTING INSTRUCTIONS
Create an alert, exit out. Create another alert, there should not be any sql in the sql input form. 

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by geido** [comment]
@AAfghahi I would like to get the test env up. Please let me know when the CI passes. Thank you

**Comment by ktmud** [comment]
In general it's preferable to always add debounce to text inputs because even if they don't trigger remote API calls, they could still trigger expensive re-rendering. Not sure if this will work, but an alternative solution here could be to re-render the modal completely when users go to a new report, which can probably be achieved by setting a uniq `key` to [the `TextAreaControl` component](https://github.com/apache/superset/blob/9b17e86b447dc36fd39c3912d3ed3baf78d3bb2e/superset-frontend/src/views/CRUD/alert/AlertReportModal.tsx#L1124-L1133).

**Review by eschutho — CHANGES_REQUESTED** [review]
small updates to how we're passing in the data

**Review by eschutho — COMMENTED** [review]
This look really good. Just a few small questions since this component is used throughout the app.

File: superset-frontend/src/views/CRUD/alert/AlertReportModal.tsx [hunk 1]
```diff
@@ -1136,7 +1136,8 @@ const AlertReportModal: FunctionComponent<AlertReportModalProps> = ({
                   maxLines={15}
                   onChange={onSQLChange}
                   readOnly={false}
-                  value={currentAlert ? currentAlert.sql : ''}
+                  initialValue={resource?.sql}
+                  key={currentAlert?.id}
```
[the remark this answers is not in the document]
Comment (AAfghahi) [reply superset-frontend/src/views/CRUD/alert/AlertReportModal.tsx:1140]: If you remove this one then if you go from one already existing alert to another already existing report, the sql shows up as blank.

**Comment by rosemarie-chiu** [comment]
@AAfghahi @eschutho I'm not sure if this is related to this fix, I found that I no longer able to see the query on the virtual dataset after this code change.

Steps:
1. In Sql editor, run a query and explore save as new dataset
2. Check the sql on the dataset and not able to see anything
3. Only able to see the query on that virutal dataset when clicking on view in sql lab


https://user-images.githubusercontent.com/52086618/139484019-9767f5cf-a036-4460-88b1-b7adee54df52.mov


I tested on other Ephemeral env and do not reproduce this, so I suspect it is caused by this PR.

✅ regression test creating new alert, edit alert sql editor field not overwrites
✅ regression test sql editor save query, update saved query, multiple query tab

**Comment by eschutho** [comment]
/testenv up ENABLE_TEMPLATE_PROCESSING=True FEATURE_ALERT_REPORTS=true

**Comment by eschutho** [comment]
> @AAfghahi @eschutho I'm not sure if this is related to this fix, I found that I no longer able to see the query on the virtual dataset after this code change.

Thanks @rosemarie-chiu! Nice catch... We'll take a look.

**Comment by AAfghahi** [comment]
@rosemarie-chiu you're amazing thank you! I made a change that catches that instance.

**Comment by eschutho** [comment]
/testenv up ENABLE_TEMPLATE_PROCESSING=True FEATURE_ALERT_REPORTS=true

**Review by eschutho — APPROVED** [review]
looks great!

**Review by rosemarie-chiu — APPROVED** [review]
LGTM! QA sign-off tested:
✅ SQL editor in sql lab
✅ SQL editor in alert
✅ SQL editor in dataset edit modal

**Comment by Renderz** [comment]
I wonder why to remove 'props.value' from TextArea, which should always render the value be given.
Suppose I want to manually change the datasource instead of typing in TextArea itself, value should be injected from props, and should also be rendered correctly.

**Pull request merged**